pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go

```diff
@@ -237,7 +239,40 @@ func TestTransitiveClosure(t *testing.T) {
 			},
 			want: wantGraph{ids: []string{"a", "b", "c"}},
 		},
-		// TODO(bowei): add error test cases.
+		{
+			name:  "nodes with error results in err",
+			cloud: mockCloud,
+			graph: func() *rgraph.Builder {
+				g := rgraph.NewBuilder()
+				addNodeToGraph(g, "a", []string{"b", "c"})
+				addNode("b", []string{"d", "e"}, nodeIsInError)
```

Comment:
we need to fix these references to non-existant nodes? the same comment applies to all of the new test cases

Reply:
done